src/components/organisms/AppRouter/AppRouter.jsx

```diff
@@ -84,7 +87,15 @@ const AppRouter = () => {
         <Route path="/v/:venueId" component={VenueLandingPage} />
         <Route path="/e/:step/:venueId" component={VenueEntrancePage} />
         <Route path="/in/:venueId/admin" component={VenueAdminPage} />
-        <Route path="/in/:venueId" component={VenuePage} />
+
+        <Route
+          path="/in/:venueId"
+          render={(props) => (
+            <RelatedVenuesProvider venueId={props.match.params.venueId}>
+              <VenuePage />
+            </RelatedVenuesProvider>
+          )}
+        />
```

Comment:
Not sure that this functional router render pattern is the right way to go here. Also not sure about including it the in `AppRouter` like this in general.

Reply:
Maybe VenuePage can have a container component and all of this declerative code can be moved there. That way this will still keep the `component` router syntax.

Reply:
That’s more or less what TemplateWrapper ends up as. It’s the bulk of stuff from VenuePage minus a bunch of legacy stuff IMO

Reply:
Superseded by `AdminRouter.tsx`'s `Provided` wrapper in 5f2ba2663620de1631552bd47e3d59a3dcab5ce2

```
          <Route path="/in/:venueId">
            <Provided withWorldUsers withRelatedVenues>
              <VenuePage />
            </Provided>
          </Route>
```